src/IO/ReadSettings.h

```diff
@@ -109,6 +109,7 @@ struct ReadSettings
 
     /// Bandwidth throttler to use during reading
     ThrottlerPtr remote_throttler;
+    ThrottlerPtr local_throttler;
```

Comment:
But sometimes you may need both, for example when some part of data is in remote storage and other on local. 
While BACKUPs is another story, I can change it to something like this:

```c
enum class ReadSettingsKind
{
    GLOBAL,
    BACKUP,
};
```

But, does it worth it?

Reply:
No, that won't simplify the code

Reply:
I agree that sometimes you need both. It all boils down to the fact that `ReadSettings` is obtained from the context (highest level) where you do not know what throttler you need. It can be even patched while we go down (e.g. by IObjectStorage). So my initial idea was to have different throttlers available in ReadSettings for different resources we want to protect (local=disk, remote=network) and on some lower level select one, depending on the Buffer type used. This way you never choose the wrong throttler.

Although, I do not insist this is the only right way. It can be don't in the way Vitaly suggests, but some care should be taken to avoid using passing remote throttles to the local buffer and vice versa. But, I'd suggest doing this in a separate PR, if we are going to change this.

Reply:
Yep, I also think that it is better to fix this separately, I'm trying to keep patches/PRs smaller to make review simpler.

And as far as I understood @vitlibar is OK with current interface after by last comment, marking this as resolved.